nilearn/decoding/decoder.py

```diff
@@ -672,7 +678,7 @@ def fit(self, X, y, groups=None):
 
         # Define the number problems to solve. In case of classification this
         # number corresponds to the number of binary problems to solve
-        y = self._binarize_y(y)
+        y = self._set_classes(y)
```

Comment:
This comment is not relevant here, but it's relevant to `decoder.py`.

At line `784`, the code:

```python
 if is_classifier(self) and (self.n_classes_ == 2):
```

I think `is_classifier` check is not necessary as `self.n_classes_` or `len(self._get_classes)` will never be 2 for a regressor.

The same is valid for line `799`.